From 36cbf8a11d43fecb7d15eae52600b26ce6db62d5 Mon Sep 17 00:00:00 2001
From: Namit Jain <namit@apache.org>
Date: Thu, 24 May 2012 20:36:54 +0000
Subject: [PATCH 065/144] HIVE-3051 JDBC cannot find metadata for tables/columns containing uppercase
 character (Navis via namit)

git-svn-id: https://svn.apache.org/repos/asf/hive/trunk@1342428 13f79535-47bb-0310-9956-ffa450edef68
(cherry picked from commit 2ea3bb8fd837de048c0ccfefedf1ade7e5501a9f)
---
 .../hadoop/hive/jdbc/HiveDatabaseMetaData.java     |    2 +-
 .../apache/hadoop/hive/jdbc/TestJdbcDriver.java    |    3 +++
 2 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/src/jdbc/src/java/org/apache/hadoop/hive/jdbc/HiveDatabaseMetaData.java b/src/jdbc/src/java/org/apache/hadoop/hive/jdbc/HiveDatabaseMetaData.java
index e59a65e..41b8048 100644
--- a/src/jdbc/src/java/org/apache/hadoop/hive/jdbc/HiveDatabaseMetaData.java
+++ b/src/jdbc/src/java/org/apache/hadoop/hive/jdbc/HiveDatabaseMetaData.java
@@ -171,7 +171,7 @@ public class HiveDatabaseMetaData implements java.sql.DatabaseMetaData {
           } else if (c == '_') {
             result.append('.');
           } else {
-            result.append(c);
+            result.append(Character.toLowerCase(c));
           }
         }
       }
diff --git a/src/jdbc/src/test/org/apache/hadoop/hive/jdbc/TestJdbcDriver.java b/src/jdbc/src/test/org/apache/hadoop/hive/jdbc/TestJdbcDriver.java
index f8103bf..56d2307 100644
--- a/src/jdbc/src/test/org/apache/hadoop/hive/jdbc/TestJdbcDriver.java
+++ b/src/jdbc/src/test/org/apache/hadoop/hive/jdbc/TestJdbcDriver.java
@@ -659,8 +659,11 @@ public class TestJdbcDriver extends TestCase {
     Map<String[], Integer> tests = new HashMap<String[], Integer>();
     tests.put(new String[]{"testhivejdbcdriver\\_table", null}, 2);
     tests.put(new String[]{"testhivejdbc%", null}, 8);
+    tests.put(new String[]{"testhiveJDBC%", null}, 8);
+    tests.put(new String[]{"testhiveJDB\\C%", null}, 0);
     tests.put(new String[]{"%jdbcdriver\\_table", null}, 2);
     tests.put(new String[]{"%jdbcdriver\\_table%", "under\\_col"}, 1);
+    tests.put(new String[]{"%jdbcdriver\\_table%", "under\\_COL"}, 1);
     tests.put(new String[]{"%jdbcdriver\\_table%", "under\\_co_"}, 1);
     tests.put(new String[]{"%jdbcdriver\\_table%", "under_col"}, 1);
     tests.put(new String[]{"%jdbcdriver\\_table%", "und%"}, 1);
-- 
1.7.0.4

